core/chains/orm.go

```diff
@@ -12,23 +12,31 @@ import (
 	"github.com/smartcontractkit/chainlink/core/services/pg"
 )
 
-// ORM manages chains and nodes.
-type ORM[I ID, C Config, N Node] interface {
-	Chain(I, ...pg.QOpt) (DBChain[I, C], error)
-	Chains(offset, limit int, qopts ...pg.QOpt) ([]DBChain[I, C], int, error)
-	CreateChain(id I, config C, qopts ...pg.QOpt) (DBChain[I, C], error)
-	UpdateChain(id I, enabled bool, config C, qopts ...pg.QOpt) (DBChain[I, C], error)
+type ChainsORM[I ID, CFG Config, C DBChain[I, CFG]] interface {
```

Comment:
Just a bit of separation and composition here for readability, and it lets us capture the DBChain type in a param so that it doesn't have to be duplicated around in the long form.